apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts

```diff
@@ -0,0 +1,450 @@
+                    // Assign the current branch only when it is not selected. This check is necessary as
+                    // opening the branch switcher would reset the current selected branch as this subscription is run everytime
+                    // branches are loaded
+
+                    if (this.currentOrganizationType === OrganizationType.Branch) {
+                        currentBranchUniqueName = this.generalService.currentBranchUniqueName;
+                        this.currentBranch = cloneDeep(response.find(branch => branch?.uniqueName === currentBranchUniqueName));
+                        this.hasTaxNumber = this.currentBranch?.addresses?.filter(address => address?.taxNumber?.length > 0)?.length > 0;
+                    } else {
+                        currentBranchUniqueName = this.activeCompany ? this.activeCompany.uniqueName : '';
+                        this.currentBranch = {
+                            name: this.activeCompany ? this.activeCompany.name : '',
+                            alias: this.activeCompany ? this.activeCompany.nameAlias || this.activeCompany.name : '',
+                            uniqueName: this.activeCompany ? this.activeCompany.uniqueName : '',
+                        };
+                    }
+                    if (this.hasTaxNumber || this.currentOrganizationType === OrganizationType.Company) {
+                        this.loadTaxDetails();
+                    }
+                }
+            } else {
+                if (this.generalService.companyUniqueName) {
+                    // Avoid API call if new user is onboarded
+                    this.store.dispatch(this.settingsBranchAction.GetALLBranches({ from: '', to: '' }));
+                }
+            }
+        });
+    }
+
+    /**
+    * Lifecycle hook which detects any changes in values
+    *
+    * @param {SimpleChanges} changes
+    * @memberof VatReportFiltersComponent
+    */
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes && this.isZimbabweCompany && changes.vatReportCurrencyCode) {
+            // Extract Currency symbol by currency code and emits
+            this.currentCurrencySymbol.emit(this.vatReportCurrencyList.filter(item => item.label === changes.vatReportCurrencyCode.currentValue).map(item => item.additional.symbol).join());
+        }
+    }
+
+    /**
+     * Loads the tax details of a company
+     *
+     * @private
+     * @memberof VatReportFiltersComponent
+     */
+    private loadTaxDetails(): void {
+        this.isTaxApiInProgress.emit(true);
+        this.gstReconcileService.getTaxDetails().pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response && response.body) {
+                this.taxes = response.body.map(tax => ({
+                    label: tax,
+                    value: tax
+                }));
+            }
+            this.isTaxApiInProgress.emit(false);
+            this.taxNumber = this.taxes[0]?.value;
```

Comment:
move this and next line under above if condition

Reply:
Done